manifests/init.pp

```diff
@@ -1,24 +1,27 @@
 #
 class logrotate (
-  String $ensure                     = present,
-  Boolean $hieramerge                = false,
-  Boolean $manage_cron_daily         = true,
-  Boolean $create_base_rules         = true,
-  Boolean $purge_configdir           = false,
-  String $package                    = 'logrotate',
-  Hash $rules                        = {},
-  Optional[Hash] $config             = undef,
-  Integer[0,23] $cron_daily_hour     = $logrotate::params::cron_daily_hour,
-  Integer[0,59] $cron_daily_minute   = $logrotate::params::cron_daily_minute,
-  Integer[0,59] $cron_hourly_minute  = $logrotate::params::cron_hourly_minute,
-  String $cron_hourly_file           = $logrotate::params::cron_hourly_file,
-  String $configdir                  = $logrotate::params::configdir,
-  String $logrotate_bin              = $logrotate::params::logrotate_bin,
-  String $logrotate_conf             = $logrotate::params::logrotate_conf,
-  Boolean $manage_package            = $logrotate::params::manage_package,
-  String $rules_configdir            = $logrotate::params::rules_configdir,
-  String $root_user                  = $logrotate::params::root_user,
-  String $root_group                 = $logrotate::params::root_group,
+  String $ensure                         = present,
+  Boolean $hieramerge                    = false,
+  Boolean $manage_cron_daily             = true,
+  Boolean $create_base_rules             = true,
+  Boolean $purge_configdir               = false,
+  String $package                        = 'logrotate',
+  Hash $rules                            = {},
+  Optional[Hash] $config                 = undef,
+  Integer[0,23] $cron_daily_hour         = $logrotate::params::cron_daily_hour,
+  Integer[0,59] $cron_daily_minute       = $logrotate::params::cron_daily_minute,
+  Integer[0,59] $cron_hourly_minute      = $logrotate::params::cron_hourly_minute,
+  String $cron_hourly_file               = $logrotate::params::cron_hourly_file,
+  Stdlib::Filemode $cron_file_mode       = $logrotate::params::cron_file_mode,
+  String $configdir                      = $logrotate::params::configdir,
+  String $logrotate_bin                  = $logrotate::params::logrotate_bin,
+  String $logrotate_conf                 = $logrotate::params::logrotate_conf,
+  Stdlib::Filemode $logrotate_conf_mode  = $logrotate::params::logrotate_conf_mode,
```

Comment:
You probably knew this already but be aware that `Stdlib::Filemode` does not support all of the variants of `File mode =>`. For instance, `a+rwx`

Reply:
Correct, it currently looks like: `Pattern[/^[0124]{1}[0-7]{3}$/]` ... which matches the permissions that are set in params. 


I think the right thing to do would be to improve Stdlib::Filemode, if someone wants to use symbolic file modes. If the folks here feel strongly that we need to support it before that time, we could just make the type `String[3]` or whatever. I don't feel like we should make our own FileMode to support it.